### PR TITLE
Implement sys_dbg_read_process_memory and sys_dbg_write_process_memory

### PR DESCRIPTION
**Pull request by TGEnigma**

### CODE REVIEW

File: rpcs3/Emu/Cell/lv2/sys_dbg.cpp [hunk 1]
```diff
@@ -1,11 +1,127 @@
...
 
+error_code sys_dbg_read_process_memory(s32 pid, u32 address, u32 size, vm::ptr<void> data)
+{
+	sys_dbg.warning("sys_dbg_read_process_memory(pid=0x%x, address=0x%llx, size=0x%x, data=*0x%x)", pid, address, size, data);
 
-logs::channel sys_dbg("sys_dbg");
+	// Todo(TGEnigma): Process lookup (only 1 process exists right now)
+	if (pid != 1)
+	{
+		return 0x80010409;
```
Comment (isJuhn) [inline rpcs3/Emu/Cell/lv2/sys_dbg.cpp]: Return named error codes instead of an unnamed constant, in this case use CELL_LV2DBG_ERROR_DEINVALIDARGUMENTS

Comment (TGEnigma) [reply rpcs3/Emu/Cell/lv2/sys_dbg.cpp]: Fixed

File: rpcs3/Emu/Cell/lv2/sys_dbg.cpp [hunk 2]
```diff
@@ -1,11 +1,126 @@
...
+	if (!vm::check_addr(data.addr(), size, vm::page_writable))
+	{
+		return CELL_EFAULT;
+	}
+
+	const u32 page_address = align_down(address, 4096);
+	const u32 page_size = align(size, 4096);
+
+	// Is the memory readable?
+	bool is_readable = true;
```
Comment (VelocityRa) [inline rpcs3/Emu/Cell/lv2/sys_dbg.cpp]:
```cpp
bool is_readable = vm::page_protect(page_address, page_size, vm::page_readable);
```

File: rpcs3/Emu/Cell/lv2/sys_dbg.cpp [hunk 3]
```diff
@@ -1,11 +1,126 @@
...
+	if (!vm::page_protect(page_address, page_size, vm::page_readable))
+	{
+		is_readable = false;
+	}
+
+	// Ensure that the memory is readable
+	if (!vm::page_protect(page_address, page_size, 0, vm::page_readable))
```
Comment (VelocityRa) [inline rpcs3/Emu/Cell/lv2/sys_dbg.cpp]:
```cpp
if (!is_readable && !vm::page_protect(page_address, page_size, 0, vm::page_readable))
```

File: rpcs3/Emu/Cell/lv2/sys_dbg.cpp [hunk 4]
```diff
@@ -1,11 +1,126 @@
...
+	if (!vm::check_addr(data.addr(), size, vm::page_readable))
+	{
+		return CELL_EFAULT;
+	}
+
+	const u32 page_address = align_down(address, 4096);
+	const u32 page_size = align(size, 4096);
+
+	// Is the memory writeable?
+	bool is_writeable = true;
```
Comment (VelocityRa) [inline rpcs3/Emu/Cell/lv2/sys_dbg.cpp]:
```cpp
bool is_writeable = vm::page_protect(page_address, page_size, vm::page_writable);
```

File: rpcs3/Emu/Cell/lv2/sys_dbg.cpp [hunk 5]
```diff
@@ -1,11 +1,126 @@
...
+	if (!vm::page_protect(page_address, page_size, vm::page_writable))
+	{
+		is_writeable = false;
+	}
+
+	// Ensure that the memory is writeable
+	if (!vm::page_protect(page_address, page_size, 0, vm::page_writable))
```
Comment (VelocityRa) [inline rpcs3/Emu/Cell/lv2/sys_dbg.cpp]:
```cpp
if (!is_writeable && !vm::page_protect(page_address, page_size, 0, vm::page_writable))
```

File: rpcs3/Emu/Cell/lv2/sys_dbg.cpp (hunk 2 above)
Comment (TGEnigma) [reply rpcs3/Emu/Cell/lv2/sys_dbg.cpp]: Fixed

**Comment by AniLeo** [comment]
Why did you change submodules

**Comment by TGEnigma** [comment]
Unintentional change, should be fixed now

**Comment by TGEnigma** [comment]
@Nekotekina Issues have been addressed. Please review again.

**Pull request merged**